### PR TITLE
Disconnect warning

### PR DESCRIPTION
**Pull request by daschuer**
Remove "QObject::disconnect: Unexpected null parameter" warning and fix bug https://bugs.launchpad.net/mixxx/+bug/1843168

### CODE REVIEW

File: src/library/dlgtrackinfo.cpp
Comment (uklotzde) [inline src/library/dlgtrackinfo.cpp]: "theCueFound" -> "loadCueFound"? We have a name for "the" cue.

File: src/library/dlgtrackinfo.cpp
Comment (uklotzde) [inline src/library/dlgtrackinfo.cpp]: A hot cue with an invalid number? What are the consequences? Can we add a comment about when this occurs and what we are doing here and why?

File: src/engine/cuecontrol.cpp
Comment (uklotzde) [inline src/engine/cuecontrol.cpp]: ~~create~~

File: src/engine/cuecontrol.cpp
Comment (uklotzde) [inline src/engine/cuecontrol.cpp]: Accessing the items of m_hotcueControls should be wrapped into a function that deals with returning nullptr if the index is out of range. Duplicating this code is highly unsafe. Creating a QList for items without a well-defined default constructor should be forbidden!

File: src/library/dlgtrackinfo.cpp
Comment (Be-ing) [reply src/library/dlgtrackinfo.cpp]: There is no "load cue". The main cue can be used that way with the option to load decks there in the Deck preferences, but it is misleading to call it a "load cue" in code.

Comment (uklotzde) [reply src/library/dlgtrackinfo.cpp]: But this flag tracks if the LOAD cue has been set during the iteration. This purpose should be reflected in the naming.

Comment (Be-ing) [reply src/library/dlgtrackinfo.cpp]: The misnamed "LOAD" cue will be renamed to MainCue in #2103.

Comment (daschuer) [reply src/library/dlgtrackinfo.cpp]: .. and this is a open discussion without a final conclusion: 
https://github.com/mixxxdj/mixxx/pull/2103#discussion_r292439905

Comment (uklotzde) [reply src/library/dlgtrackinfo.cpp]: I'm just asking for a consistent naming, choose whatever you want.

Comment (daschuer) [reply src/library/dlgtrackinfo.cpp]: I am unsure how to continue here. I would prefer "The Cue" here and in #2103, because this is the name we have used since introducing hot cues. But I can also live with MainCue, if we all agree to this name. 
I can use here loadCueFound and it can be changed in #2103 to the new name or introduce it here. Wat do you think?

Comment (uklotzde) [reply src/library/dlgtrackinfo.cpp]: Just leave it as is. I didn't plan to start a fruitless discussion. It was just a minor remark.

File: src/engine/cuecontrol.cpp
Comment (uklotzde) [inline src/engine/cuecontrol.cpp]: Ok, the *defaultValue* is defined:
https://doc.qt.io/qt-5/containers.html#default-constructed-value
By omitting it we avoid confusion.

**Review by uklotzde — CHANGES_REQUESTED** [review]
Test failures

**Comment by daschuer** [comment]
I have already looked into the test failure.
It fails because the default cue at 0:00 is lost in the test case.

This discovers an other issue, our double cue point storage, one in the list of cues, the other in the track object. 

If the list of cues has no cue set, the track object cue is used, which defaults to 0:00 instead of invalid. Many implications. 

Would it be ok to store 0:00 in the cue table as well in this case? 

Has the cue column in the library any value?

**Comment by daschuer** [comment]
Should we allow to delete the cue point, or should we force it to 0:00? We initial use 0:00 if no cue is set.

File: src/library/dlgtrackinfo.cpp
Comment (uklotzde) [inline src/library/dlgtrackinfo.cpp]: These new lies don't help

Comment (daschuer) [reply src/library/dlgtrackinfo.cpp]: Why do you think that? Do you have any idea to do it better?

Comment (uklotzde) [reply src/library/dlgtrackinfo.cpp]: Sorry, it was too late ;) The diff just shows two empty lines that could be removed to improve readability.

Comment (uklotzde) [reply src/library/dlgtrackinfo.cpp]: I'll delete those empty lines myself after merging the PR ;)

**Review by uklotzde — APPROVED** [review]

**Comment by uklotzde** [comment]
I know from my own experience that the cue control code is brittle and would benefit from a rewrite. Thanks for ironing out some of its rough edges. Hopefully, we didn't miss any edge cases.

LGTM

**Comment by uklotzde** [comment]
Causes many merge conflicts with master! @daschuer Are you able to resolve these?

**Comment by Be-ing** [comment]
> I know from my own experience that the cue control code is brittle and would benefit from a rewrite.

We should probably hold off on that until #2194 is merged (unless it somehow becomes necessary to get those new features working correctly).

**Pull request merged**
